build.gradle

```diff
@@ -320,6 +321,7 @@ project(':iceberg-flink') {
     compile project(':iceberg-parquet')
     compile project(':iceberg-hive-metastore')
 
+    compileOnly "org.apache.flink:flink-connector-base"
```

Comment:
The interfaces (such as `SplitReader`, `RecordsWithSplitIds`, `SourceReaderOptions`) from this jar  are stable enough to expose for downstream project like iceberg , ha ?   I rise this Q because I don't see it's marked with a `Public` annotation and seems don't have any compatibility guarantee ?

Reply:
`SplitReader`, `RecordsWithSplitIds` are part of the core API that source implementation needs to extend/implement from.

Reply:
I did a little research and saw Stephan recently merged the `@PublicEvolve` annotation to all 3 interfaces that you pointed out: https://issues.apache.org/jira/browse/FLINK-22358

Reply:
It would be too restrictive to only depend on `@Public` interfaces. FLIP-27 connectors are becoming the default and these interface should mature soon though. Iceberg project will need a story to support multiple Flink versions at some point.